### PR TITLE
main/ansible-core: depend on python-passlib

### PR DESCRIPTION
**Pull request by jcgruenhage**
Packages python-passlib and makes ansible-core depend on it. While ansible-core does somewhat work without passlib, most things around hashing don't work without it.

### CODE REVIEW

File: main/ansible-core/template.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "python-cryptography",
     "python-jinja2",
     "python-packaging",
+    "python-passlib",
```
Comment (nekopsykose) [inline main/ansible-core/template.py:14]: need to bump pkgrel for this

File: main/python-bcrypt/template.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+url = "https://github.com/pyca/bcrypt"
+source = f"{url}/archive/refs/tags/{pkgver}.tar.gz"
+sha256 = "36c757ad7a317627256b7f0be570b16ee1639c6faf398f9f7a7bd4f21bd289ee"
+
+
+def prepare(self):
+    from cbuild.util import cargo
+
+    with self.pushd("src/_bcrypt"):
+        cargo.Cargo(self).vendor()
```
Comment (nekopsykose) [inline main/python-bcrypt/template.py:27]:
```suggestion
    cargo.Cargo(self, wrksrc="src/_bcrypt").vendor()
```

then delete the pushd below in init_build

File: main/python-bcrypt/template.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+def init_build(self):
+    from cbuild.util import cargo
+
+    with self.pushd("src/_bcrypt"):
+        renv = cargo.get_environment(self)
+        self.make_env.update(renv)
+
+
+def post_install(self):
+    self.install_license("LICENSE")
```
Comment (nekopsykose) [inline main/python-bcrypt/template.py:39]: apache doesn't have to be installed

File: main/python-bcrypt/template.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+pkgname = "python-bcrypt"
+pkgver = "4.2.0"
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "python-setuptools-rust",
+    "python-cffi",
+    "python-build",
```
Comment (nekopsykose) [inline main/python-bcrypt/template.py:8]: sort

File: main/python-bcrypt/template.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "python-installer",
+    "cargo",
+]
+makedepends = ["python-devel", "rust-std"]
+depends = ["python-cffi"]
```
Comment (nekopsykose) [inline main/python-bcrypt/template.py:13]: i don't see cffi used anywhere in the codebase, or for build, or for tests, and they pass without it. are you sure it's used?

File: main/python-bcrypt/template.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+checkdepends = [*depends, "python-pytest"]
+pkgdesc = "Modern password hashing for software and servers"
```
Comment (nekopsykose) [inline main/python-bcrypt/template.py:15]:
```suggestion
pkgdesc = "Bcrypt password hashing for python"
```

File: main/python-bcrypt/template.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "cargo",
+    "python-build",
+    "python-installer",
+    "python-setuptools-rust",
+]
+makedepends = ["python-devel", "rust-std"]
+checkdepends = ["python-pytest"]
```
Comment (nekopsykose) [inline main/python-bcrypt/template.py:12]:
```suggestion
checkdepends = ["python-pytest"]
depends = ["python"]
```

needed for leaf python package

File: main/python-bcrypt/template.py (hunk 5 above)
Comment (jcgruenhage) [reply main/python-bcrypt/template.py]: I really just shouldn't look at how others package something. cffi was used *until* version 4, which is when python-bcrypt was rewritten in rust.

Comment (jcgruenhage) [reply main/python-bcrypt/template.py]: (Thanks for catching this.)

**Comment by nekopsykose** [comment]
thanks, also made the passlib tests parallel so they take less forever

**Pull request merged**